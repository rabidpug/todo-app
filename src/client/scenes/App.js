import { Layout, } from 'antd';
import React from 'react';
import { Switch, } from 'react-router-dom';
import actionMenu from 'Routes/actionMenu';
import content from 'Routes/content';
import { hot, } from 'react-hot-loader';
import navMenu from 'Routes/navMenu';
import { renderRoutes, } from 'react-router-config';
import styles from './App.scss';

const { Content, } = Layout;

const App = () => (
  <Layout className={ styles.layoutStyle }>
    {renderRoutes( navMenu )}
    <Layout>
      <Switch>{renderRoutes( actionMenu )}</Switch>
      <Content className={ styles.bodyStyle }>{renderRoutes( content )}</Content>
    </Layout>
  </Layout>
);

export default hot( module )( App );
