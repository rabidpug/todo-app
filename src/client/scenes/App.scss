body {
  background-color: #d9d9d9;
}

.bodyStyle {
  background: #f5f5f5;
  margin: 12px 8px;
  min-height: 280;
  overflow: auto;
  padding: 24;
}

.layoutStyle {
  height: 100vh;
  overflow: hidden;
}
