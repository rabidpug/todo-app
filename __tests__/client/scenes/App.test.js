import App from 'Scenes/App';
import React from 'react';
import { shallow, } from 'enzyme';
import { shallowToJson, } from 'enzyme-to-json';

describe(
  'App', () => {
    it(
      'Should render the App', () => {
        const wrapper = shallow( <App /> );

        expect( shallowToJson( wrapper ) ).toMatchSnapshot();
      }
    );
  }
);
