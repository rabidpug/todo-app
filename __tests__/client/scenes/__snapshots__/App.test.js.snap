// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Should render the App 1`] = `
<Adapter
  className="layoutStyle"
>
  <Switch>
    <Route
      key="0"
      path="*"
      render={[Function]}
    />
  </Switch>
  <Adapter>
    <Switch>
      <Switch>
        <Route
          exact={true}
          key="0"
          path="/todo"
          render={[Function]}
        />
        <Route
          key="1"
          path="*"
          render={[Function]}
        />
      </Switch>
    </Switch>
    <Adapter
      className="bodyStyle"
    >
      <Switch>
        <Route
          exact={true}
          key="0"
          path="/"
          render={[Function]}
        />
        <Route
          exact={true}
          key="1"
          path="/welcome"
          render={[Function]}
        />
        <Route
          exact={true}
          key="2"
          path="/bye"
          render={[Function]}
        />
        <Route
          exact={true}
          key="3"
          path="/todo"
          render={[Function]}
        />
      </Switch>
    </Adapter>
  </Adapter>
</Adapter>
`;
